metpy/calc/tools.py

```diff
@@ -1126,3 +1135,51 @@ def _process_deriv_args(f, kwargs):
         raise ValueError('Must specify either "x" or "delta" for value positions.')
 
     return n, axis, delta
+
+
+@exporter.export
+def parse_angle(input_dir, deg2rad=False):
+    """Calculate the meteorological angle from directional text
+
+    Works for abbrieviations or whole words (E -> 90 | South -> 180)
+    and also is able to parse 22.5 degreee angles such as ESE/East South East
+
+    Parameters
+    ----------
+    input_dir : string or array-like strings
+        Directional text such as west, [south-west, ne], etc
+    deg2rad : boolean, optional
+        If true, returns a value in radian, else returns a value in degrees
+
+    Returns
+    -------
+    angle
+        The angle in degrees or radians
+    """
+    if len(input_dir) > 1 and not isinstance(input_dir, str):
+        deg = [parse_angle(single_dir) for single_dir in input_dir]
```

Comment:
Elegant, but need to make sure you pass `deg2rad` here--assuming we keep it.

Reply:
I don't think I do or else I would be converting it twice since `if deg2rad` is outside the `if else` statements (the test, `def test_parse_angle_mix_multiple():` confirms this.

Reply:
Oops. 🐑  You're right.